chroma-agent/chroma_agent/agent_daemon.py

```diff
@@ -50,112 +40,33 @@ def boot_time(self):
-                victims.append(pid)
-                os.kill(pid, signal.SIGKILL)
-
-    n = 0
-    while True:
-        for pid in victims:
-            if not os.path.exists("/proc/%s" % pid):
-                sys.stderr.write("Process %s terminated successfully\n")
-                victims.remove(pid)
-
-        if victims:
-            n += 1
-
-            if n > timeout:
-                raise RuntimeError("Failed to kill orphan processes %s after %s seconds" % (victims, timeout))
-
-            time.sleep(1)
-        else:
-            break
-
-
 def main():
-    """Daemonize and handle unexpected exceptions"""
-    parser = argparse.ArgumentParser(description="Intel® Manager for Lustre* software Agent")
-    parser.add_argument("--foreground", action="store_true")
+    """handle unexpected exceptions"""
+    parser = argparse.ArgumentParser(
+        description="Intel® Manager for Lustre* software Agent")
+
     parser.add_argument("--publish-zconf", action="store_true")
-    parser.add_argument("--pid-file", default = "/var/run/chroma-agent.pid")
     args = parser.parse_args()
 
     # FIXME: at startup, if there is a PID file hanging around, find any
     # processes which are children of that old PID, and kill them: prevent
     # orphaned processes from an old agent run hanging around where they
     # could cause trouble (think of a 2 hour mkfs)
 
-    if not args.foreground:
-        if os.path.exists(args.pid_file):
-            pid = None
-            try:
-                pid = int(open(args.pid_file).read())
-                os.kill(pid, 0)
-            except (ValueError, OSError, IOError):
-                # Not running, delete stale PID file
-                sys.stderr.write("Removing stale PID file\n")
-                try:
-                    os.remove(args.pid_file)
-                    os.remove(args.pid_file + ".lock")
-                except OSError, e:
-                    import errno
-                    if e.errno != errno.ENOENT:
-                        raise e
-
-                if pid is not None:
-                    kill_orphans_of(pid)
-            else:
-                # Running, we should refuse to run
-                raise RuntimeError("Daemon is already running (PID %s)" % pid)
-        else:
-            if os.path.exists(args.pid_file + ".lock"):
-                sys.stderr.write("Removing stale lock file\n")
-                os.remove(args.pid_file + ".lock")
-
-        signal.signal(signal.SIGHUP, signal.SIG_IGN)
-        context = DaemonContext(pidfile = PIDLockFile(args.pid_file))
-        context.open()
-
-        daemon_log_setup()
-        console_log_setup()
-        daemon_log.info("Starting in the background")
-    else:
-        context = None
-        daemon_log_setup()
-        daemon_log.addHandler(logging.StreamHandler())
-
-        console_log_setup()
+    signal.signal(signal.SIGHUP, signal.SIG_IGN)
+
+    daemon_log_setup()
```

Comment:
maybe not as part of this patch but this gives us a good opportunity to consolidate the Daemon and console logs together, I was never convinced that we needed them to exist separately anyway

Reply:
Yeah, I agree. Even beyond that we could dump the external logs altogether and just write everything to the journal.

Wonder if this will take care of the mismatched timestamps as well?

Reply:
Was thinking of that happening in separate patch anyway.